org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/SemverParser.java

```diff
@@ -0,0 +1,540 @@
+package org.hl7.fhir.utilities;
+import java.util.*;
+
+public class SemverParser {
+  private final String input;
+  private int pos;
+  private final List<String> parts;
+  private final boolean allowWildcards;
+  private final boolean requirePatch;
+
+  public SemverParser(String input) {
+    this(input, false, true);
+  }
+
+  public SemverParser(String input, boolean allowWildcards) {
+    this(input, allowWildcards, true);
+  }
+
+  public SemverParser(String input, boolean allowWildcards, boolean requirePatch) {
+    this.input = input;
+    this.pos = 0;
+    this.parts = new ArrayList<>();
```

Comment:
This feels like it exposes the implementation details of SemverParser. In the main code, we should really just care about things like major, minor, patch, etc. and have those populated when the parse is finished.